src/Sparrow/ByteString.cs

```diff
@@ -737,10 +817,19 @@ private SegmentInformation AllocateSegment(int size)
 
         private void AllocateExternalSegment(int size)
```

Comment:
Are there any promises regarding the `size` ? Is it always power of 2 (or rather, power of 64K) ?